### PR TITLE
Anonymous use

### PR DESCRIPTION
**Pull request by rodrigok**
@RocketChat/core 

# TODO
- [x] Reload current room after user registration
- [x] List all default rooms

image

Refs: https://github.com/RocketChat/Rocket.Chat/issues/604

### CODE REVIEW

File: packages/rocketchat-ui-master/master/main.coffee [hunk 1]
```diff
@@ -102,7 +102,7 @@ Template.main.helpers
 		return ready
 
 	hasUsername: ->
-		return Meteor.userId()? and Meteor.user().username?
+		return (Meteor.userId()? and Meteor.user().username?) || RocketChat.authz.hasAllPermission('preview-c-room')
```
Comment (rodrigok) [inline packages/rocketchat-ui-master/master/main.coffee]: This will prevent the username screen for normal users too

**Comment by sampaiodiego** [comment]
even I approved it, I can still see odd behaviors in 2 scenarios:

1)
 - enter as guest
 - click on "register or login to send messages"
 - log in or register
 - then hit logout
 - you'll be redirect to login page and you have to refresh to use as guest again

2)
 - enter as guest
 - log in
 - refresh the page
 - you'll be still logged in
 - then hit logout
 - you'll see the chat as guest but will not see any channels listed on left
   - if you go to "more channels", it'll display not display the default one you should be looking at (#general in my test)

**Comment by rodrigok** [comment]
The number 1 was fixed.
The number 2 is more complex, it's related with cache

**Comment by engelgabriel** [comment]
@rodrigok should we fix the conflicts and merge this? 
Can we open an issue for the outstanding bug?

**Pull request merged**